pyaerocom/io/read_mscw_ctm.py

```diff
@@ -252,82 +393,68 @@ def read_var(self, var_name, ts_type=None, **kwargs):
-
-        # Find path to file based on ts_type
-        filepath = ''
-        if self.data_dir is not None:
-            filename = ''
-            if ts_type == 'monthly':
-                filename = 'Base_month.nc'
-            elif ts_type == 'hourly':
-                filename = 'Base_hour.nc'
-            elif ts_type == 'daily':
-                filename = 'Base_day.nc'
-            elif ts_type == 'yearly':
-                filename = 'Base_fullrun.nc'
-            filepath = os.path.join(self.data_dir, filename)
-            if not os.path.isfile(filepath):
-                raise FileNotFoundError('Could not find file: {}'.format(filepath))
-        elif self.filepath is not None:
-            filepath = self.filepath
-            if ts_type is None:
-                ts_type = ts_type_from_filename(os.path.split(filepath)[-1])
-
-        if var_name in self.AUX_REQUIRES:
-            temp_cubes = []
-            for aux_var in self.AUX_REQUIRES[var_name]:
-                temp_cubes.append(self.read_var(aux_var, ts_type=ts_type))
-            aux_func = self.AUX_FUNS[var_name]
-            cube = aux_func(*temp_cubes)
-            gridded = GriddedData(cube, var_name=var_name, ts_type=ts_type, computed=True)
-        else:
-            try:
-                emep_var = var_map[var_name]
-            except KeyError:
-                raise VarNotAvailableError('Variable {} not in EMEP mapping.'.format(var_name))
-            EMEP_prefix = emep_var.split('_')[0]
-            data = xr.open_dataset(filepath)[emep_var]
-            data.attrs['long_name'] = var_name
-            data.time.attrs['long_name'] = 'time'
-            data.time.attrs['standard_name'] = 'time'
-            data.attrs['units'] = self.preprocess_units(data.units, EMEP_prefix)
-            cube = data.to_iris()
-            if ts_type == 'hourly':
-                cube.coord('time').convert_units('hours since 1900-01-01')
-            gridded = GriddedData(cube, var_name=var_name, ts_type=ts_type,
-                                  check_unit=False,
-                                  convert_unit_on_init=False)
-
-            if EMEP_prefix in ['WDEP', 'DDEP']:
-                implicit_to_explicit_rates(gridded, ts_type)
+        if var.is_deposition:
+            implicit_to_explicit_rates(gridded, ts_type)
 
         # At this point a GriddedData object with name gridded should exist
 
         gridded.metadata['data_id'] = self.data_id
-        gridded.metadata['from_files'] = filepath # ReadGridded cannot concatenate several years of data if this is missing
+        gridded.metadata['from_files'] = self.filepath
 
         # Remove unneccessary metadata. Better way to do this?
         for metadata in ['current_date_first', 'current_date_last']:
             if metadata in gridded.metadata.keys():
                 del(gridded.metadata[metadata])
         return gridded
 
+    def _read_var_from_file(self, var_name_aerocom, ts_type):
```

Comment:
Should we have docstrings for private methods as well? It can be difficult to understand the difference between all the methods called something with read or load.

Reply:
Absolutely! Thanks for pointing this out :)